tests/executables_src/testDataValidityPropagation.cc

```diff
@@ -0,0 +1,552 @@
+    }
+  }
+};
+
+struct TriggerModule : ctk::ApplicationModule {
+  using ctk::ApplicationModule::ApplicationModule;
+  ctk::ScalarOutput<int> o1{this, "o1", "", ""};
+  void mainLoop() override {
+    // do nothing, we trigger o1 manually in the tests
+    //    while(true) {
+    //      // using boost sleep helps for the interruption point
+    //      boost::this_thread::sleep_for(boost::chrono::milliseconds(200));
+    //      o1=1;
+    //      o1.write();
+    //    }
+  }
+};
+
+template<typename ModuleT>
+struct TestApplication1 : ctk::Application {
+  TestApplication1() : Application("testSuite") {}
+  ~TestApplication1() { shutdown(); }
+
+  void defineConnections() { findTag(".*").connectTo(cs); }
+
+  ModuleT mod{this, "m1", ""};
+  ctk::ControlSystemModule cs;
+
+  TriggerModule m2{this, "m2", ""};
+  ctk::ConnectingDeviceModule dev{
+      this, "(ExceptionDummy?map=testDataValidityPropagation.map)", "/m2/o1", &initialiseDev};
+  std::atomic_bool deviceError = false;
+  static void initialiseDev(ChimeraTK::DeviceModule*) {
+    bool err = ((TestApplication1&)Application::getInstance()).deviceError;
+    if(err) {
+      throw ChimeraTK::runtime_error("device is not ready.");
+    }
+  }
+};
+
+//BOOST_AUTO_TEST_SUITE(dataValidityPropagation)
+
+struct TestApplication3 : ctk::Application {
+  constexpr static char const* ExceptionDummyCDD = "(ExceptionDummy?map=testDataValidityPropagation2.map)";
+  TestApplication3() : Application("testPartiallyInvalidDevice") {}
+  ~TestApplication3() { shutdown(); }
+
+  TriggerModule m2{this, "m2", ""};
+
+  ctk::ControlSystemModule cs;
+  ctk::ConnectingDeviceModule device1{this, ExceptionDummyCDD, "/m2/o1"};
+
+  //ctk::DeviceModule device1{this, ExceptionDummyCDD};
+  //  void defineConnections() override {
+  //    // use manual connection setup instead of ConnectingDeviceModule.
+  //    // note, differently from automatic connection setup, registers contain . and not / as separator
+  //    auto pollInput1 = device1("dev.i1", typeid(int), 1, ChimeraTK::UpdateMode::poll);
+  //    auto pollInput2 = device1("dev.i2", typeid(int), 1, ChimeraTK::UpdateMode::poll);
+  //    auto trigger = m2("o1");
+  //    pollInput1[trigger] >> cs("dev.i1", typeid(int));
+  //    pollInput2[trigger] >> cs("dev.i2", typeid(int));
+  //  }
+};
+/**
+ *  tests the ExceptionDummyPollDecorator of the ExceptionDummyBackend, which provides a way for
+ *  forcing individual (poll-type) device outputs to DataValidity=faulty
+ */
+BOOST_AUTO_TEST_CASE(testDataValidity_exceptionDummy) {
+  TestApplication3 app;
+  ctk::TestFacility test;
+  //app.dumpConnections();
+
+  auto devI1 = test.getScalar<int>("/dev/i1");
+  auto devI2 = test.getScalar<int>("/dev/i2");
+  test.runApplication();
+  // it seems that the exceptionDummy is created only when app starts.
+  auto exceptionDummy =
+      boost::dynamic_pointer_cast<ctk::ExceptionDummy>(app.device1.getDeviceModule().device.getBackend());
+  assert(exceptionDummy);
```

Comment:
unnecessary, since the boost::shared_ptr does it when dereferencing it...

Reply:
Yes, but debugging was easier like this.